tests/app_test.rb

```diff
@@ -0,0 +1,63 @@
+ENV['RACK_ENV'] = 'test'
+
+require_relative '../app'
+require 'test/unit'
+require 'rack/test'
+require 'pry'
+
+
+configure do
+  DataMapper::setup(:default, "sqlite3:test.db")
+  DataMapper.finalize
+  DataMapper.auto_migrate!
+end
+
+class ApplicationControllerTest < Test::Unit::TestCase
+  include Rack::Test::Methods
+
+  def app
+    @app ||= ApplicationController.new
+  end
+
+  def test_home_page
+    get '/'
+    last_response.body
+    assert last_response.ok?
+  end
+
+  def test_show_form
+    get '/show'
```

Comment:
This should be called 

get `/details/new` and the corresponding template should be `new.erb` instead of `api_form`.